contracts/scripts/docker_get_genesis_hash.ts

```diff
@@ -0,0 +1,146 @@
+import { executeCommand } from "./docker_utils";
+import path from "path";
+import { Wallet, utils, ethers, BigNumber } from "ethers";
+import assert from "assert";
+import fs from "fs";
+import rollupJson from "../deployments/localhost/Rollup.json";
+import sequencerInboxJson from "../deployments/localhost/SequencerInbox.json";
+
+const ROOT_DIR = path.join(__dirname, "../../");
+const SPECULAR_DATADIR =
+  process.env.SPECULAR_DATADIR || path.join(ROOT_DIR, "specular-datadir");
+
+const sequencerPrivateKeyPath =
+  ROOT_DIR + "clients/geth/specular/data/keys/sequencer.prv";
+const validatorPrivateKeyPath =
+  ROOT_DIR + "clients/geth/specular/data/keys/validator.prv";
+
+const l2Provider = new ethers.providers.JsonRpcProvider(
+  "http://localhost:4011"
+);
+
+const l1Provider = new ethers.providers.JsonRpcProvider(
+  "http://localhost:8545"
+);
+
+// Setup signers
+export async function setupSigners(
+  sequencerPrivateKeyPath: string,
+  validatorPrivateKeyPath: string
+) {
+  const sequencerPrivateKey = fs.readFileSync(sequencerPrivateKeyPath, "utf8");
+  const sequencerSigner = new Wallet(sequencerPrivateKey, l2Provider);
+
+  const validatorPrivateKey = fs.readFileSync(validatorPrivateKeyPath, "utf8");
+  const validatorSigner = new Wallet(validatorPrivateKey, l2Provider);
+
+  return {
+    sequencerSigner,
+    validatorSigner,
+  };
+}
+
+// Initialize contracts and event filters
+function initializeContracts(
+  sequencerContractAddress: string,
+  sequencerContractAbi: any,
+  rollupContractAddress: string,
+  rollupContractAbi: any
+) {
+  const sequencerContract = new ethers.Contract(
+    sequencerContractAddress,
+    sequencerContractAbi,
+    l1Provider
+  );
+  const rollupContract = new ethers.Contract(
+    rollupContractAddress,
+    rollupContractAbi,
+    l1Provider
+  );
+
+  const appendTxFilter = sequencerContract.filters.TxBatchAppended();
+  const assertionCreatedFilter = rollupContract.filters.AssertionCreated();
+  const assertionConfirmedFilter = rollupContract.filters.AssertionConfirmed();
+
+  return {
+    sequencerContract,
+    rollupContract,
+    appendTxFilter,
+    assertionCreatedFilter,
+    assertionConfirmedFilter,
+  };
+}
+
+// Start L1 & L2
+async function startChains() {
+  console.log("In docker_start, this is ROOT_DIR: ", ROOT_DIR);
+  console.log("In docker_start, this is SPECULAR_DATADIR: ", SPECULAR_DATADIR);
+  const command = `docker-compose -f ${ROOT_DIR}docker/services/docker-compose-l1hardhat.yml up -d --build`;
```

Comment:
should be integration test docker compose file

Reply:
Unused code is removed